src/blueprint/from_config.rs

```diff
@@ -24,13 +25,34 @@ use crate::{blueprint, config};
 
 type Valid<A> = ValidDefault<A, String>;
 
+const RESTRICTED_ROUTES: [&str; 2] = ["/", "/graphql"];
+
+fn validate_route(config: &Config) -> Result<(), ValidationError<String>> {
+  if let Some(enable_graphiql) = &config.server.enable_graphiql {
+    if RESTRICTED_ROUTES.contains(&enable_graphiql.as_str()) {
+      CLIError::new(
+          &format!("Cannot use restricted routes '{}' for enabling graphiql", enable_graphiql)
+    ).trace(vec!["schema".into(), "server".into(), "enableGraphiql".into()]);
+
+    return Err(
+      ValidationError::new(
+        format!("Cannot use restricted routes '{}' for enabling graphiql", enable_graphiql)
+      )
+      );
+    }
+  }
+  Ok(())
```

Comment:
Can simplify this logic.

Reply:
ok i will simply it more, we need to throw error with trace right , if  only validation error works then another idea is to do validation in config_blueprint function it self

Reply:
You can perform a `Valid::fail` like we have done in other places.